ethereum_history_api/circuits/lib/src/rlp/decode.nr

```diff
@@ -57,21 +57,29 @@ pub fn decode_to_rlp_header<N>(arr: [u8; N]) -> RlpHeader {
     if (prefix < 0x80) { // single byte
         RlpHeader { offset: 0, length: 1, data_type: STRING }
     } else if (prefix < 0xb8) { // 0-55 byte string
+        let offset = 1;
+        let length = wrapping_sub(prefix, 0x80) as u64;
+        assert(offset + length <= N, "Length of short string exceeds input length");
+
```

Comment:
You can resue length below and inline length

Reply:
Fixed in new commit